index.d.ts

```diff
@@ -0,0 +1,4 @@
+
+type ReturnInnerFuction = (relativeUrl: string, params?: Record<string, unknown>) => string
+
+export default function urlFactory(baseUrl?: string, baseParams?: Record<string, unknown>): ReturnInnerFuction;
```

Comment:
the change itself looks good but you should add the `types` key and extend the `files` key in package.json ([example](https://github.com/Patreon/nion/blob/master/package.json#L7-L14))